### ISSUE TITLE
Support Nullish Coalescing in Rules

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#1052]
Spinning off of #1033

Rules that need to have support added for optional chaining.
This list was based on a simple grep for `LogicalExpression`, I haven't validated anything beyond that. There are probably extra rules that could do with support as well that don't check `LogicalExpression`.

- [x] no-extra-parens (issue: #1155)
    - Same as for optional chaining (#1051) - this one will be a huge pain, and I believe it will involve us duplicating the base rule. The logic is relatively complex, so this is a non-trivial task.
- [x] ~~no-misused-promises~~ this is already
- [x] no-unnecessary-condition (#1148)
- [x] strict-boolean-expressions (#1275)
- [x] unbound-method (#1440)

**Comment by 7sDream** [comment]
Do we really need `strict-boolean-expressions` to support Nullish Coalescing?

Example: 

```ts
export const something: string = process.env.VAR ?? 'default value';
```

will raise a

```plain
  1:34  error  Unexpected non-boolean in conditional  @typescript-eslint/strict-boolean-expressions
```

If I enable this rule, even if I set `allowNullable ` and `ignoreRhs` to true.

And I even can't think up a right way to fix this error by edit my code if I want to keep use `??`.

---

BTW: The [document of `strict-boolean-expressions`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md) don't mention that it will check `??`.

**Comment by bradzacher** [comment]
@7sDream 

You misunderstand what "support" means. It means that the rule should be upgraded to either (a) lint the operator appropriately, or (b) ignore the operator appropriately.

In the case of `strict-boolean-expressions`, it is definitely (b). If you click through to the pr listed against strict boolean expressions, you'll see the fix.

**Comment by bradzacher** [comment]
This should now all be supported by ESLint core.
